libmscore/undo.cpp

```diff
@@ -596,13 +596,30 @@ void Score::undoChangeClef(Staff* ostaff, Segment* seg, ClefType st)
 
 static Element* findLinkedVoiceElement(Element* e, Staff* nstaff)
       {
+      Excerpt* se = e->score()->excerpt();
+      Excerpt* de = nstaff->excerpt();
+      int strack = e->track();
+      int dtrack = nstaff->idx() * VOICES + e->voice();
+
+      if (se)
+            strack = se->tracks().key(strack);
+
+      if (de) {
+            QList<int> l = de->tracks().values(strack);
+            for (int i : l) {
+                  if (nstaff->idx() * VOICES <= i && (nstaff->idx() + 1) * VOICES > i) {
+                        dtrack = i;
```

Comment:
The function tries to find an element in a staff, even when the element is in a voice not linked to that staff.
Lets say it's called with an element e of voice 2 in the master score (<i>e->track() == 1</i>) for a staff nstaff in a voice 1 part (where e is not linked to). In this case we initialize <i>strack=dtrack=1</i>. Since in a voice 1 part (track==0) nothing is mapped to track 1, <i>l</i> is an empty list and with <i>dtrack=1</i> the function will return a null pointer and cause a crash.
This happens when you have a voice 1 part and a voice 2 part and try to change the duration of something in voice 2 in the master score.